### PR TITLE
Data Regulation API documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Removed the outdated API documentation and replaced it with the API docs for the Data Regulation service.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: rudderstack-api/data-regulation-api.mdx [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+# slug: "/docs/rudderstack-api/user-suppression"
+title: "Data Regulation API"
+description: >-
+  Detailed technical description of the RudderStack Data Regulation API for
+  suppressing and deleting user data.
+---
+
+# Data Regulation API
+
+RudderStack's Data Regulation API allows you to delete the collected data, suspend data collection, or do both - for a particular user identified by a `userId`.
```
Comment (lvrach) [inline rudderstack-api/data-regulation-api.mdx:11]: Actually we support:
* suspend data collection
* suspend data collection and delete data

You can not delete data while  data collection is not suspended

File: rudderstack-api/data-regulation-api.mdx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+* **Sample Request**
+
+```bash
+curl --location --request POST 'https://api.rudderstack.com/v2/regulations' \
+--header 'Authorization: Bearer 22aJ7iEeWud78wDaR6dbAe29Nw2' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "regulationType": "suppress",
+    "sourceIds": [
```
Comment (lvrach) [inline rudderstack-api/data-regulation-api.mdx:112]: tip: they can use `https://api.rudderstack.com/v2/sources` to get source ids

File: rudderstack-api/data-regulation-api.mdx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+---
+
+# Data Regulation API
+
+RudderStack's Data Regulation API allows you to do the following for a particular user:
+
+* Suspend data collection
+* Suspend data collection and delete data
+
+This guide covers the data regulation feature in detail as well as details the various endpoints of the Data Regulation API.
```
Comment (saurav-malani) [inline rudderstack-api/data-regulation-api.mdx:16]: This line isn't very clear. Can you please re-phrase it?

**Review by saurav-malani — CHANGES_REQUESTED** [review]

File: rudderstack-api/data-regulation-api.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply rudderstack-api/data-regulation-api.mdx:111]: mentioned this as a tip.

File: rudderstack-api/data-regulation-api.mdx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+You can set these **optional** fields to specify the sources/destinations from which the data should be suppressed or deleted. 
+
+<div class="warningBlock">
+
+<ul>
+<li>Set the <code class="inline-code">sourceIds</code> field when setting the <code class="inline-code">regulationType</code> to <code class="inline-code">suppress</code>.</li>
+<li>Set the <code class="inline-code">destinationIds</code> field when setting the <code class="inline-code">regulationType</code> to <code class="inline-code">suppress_with_delete</code>.</li>
+</ul>
+
+Avoid specifying both <code class="inline-code">sourceIds</code> and <code class="inline-code">destinationIds</code>.
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply rudderstack-api/data-regulation-api.mdx]: thanks, removed this block to avoid any confusion.

**Review by saurav-malani — APPROVED** [review]

**Review by chandumlg — APPROVED** [review]

**Pull request merged**